### PR TITLE
Feat: Add custom modal for user feedback

### PR DESCRIPTION
**Pull request by Mikehrn**
Screenshot 2024-10-02 at 12 28 22

### CODE REVIEW

File: packages/frontend-2/components/feedback/Dialog.vue [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  triggerNotification({
+    type: ToastNotificationType.Success,
+    title: 'Thank you for your feedback!'
+  })
+
+  mixpanel.track('Feedback Sent', {
+    feedback: feedback.value
+  })
+
+  await sendWebhook('https://hooks.zapier.com/hooks/catch/12120532/2m4okri/', {
```
Comment (fabis94) [inline packages/frontend-2/components/feedback/Dialog.vue:69]: Is this OK being hardcoded? All of our deployments are going to use this

File: packages/frontend-2/components/feedback/Dialog.vue [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+const { handleSubmit } = useForm<FormValues>()
+
+const feedback = ref('')
+
+const dialogButtons = computed((): LayoutDialogButton[] => [
+  {
+    text: 'Send',
+    props: { color: 'primary' },
+    onClick: () => {
+      onSubmit()
```
Comment (fabis94) [inline packages/frontend-2/components/feedback/Dialog.vue:50]: Doesn't make sense that there's `useForm` and yet you're not actually using a `<form/>`. Use the on-submit prop of LayoutDialog to properly use form submission for this and convert the button to a submit button.

**Review by fabis94 — CHANGES_REQUESTED** [review]

File: packages/frontend-2/components/feedback/Dialog.vue (hunk 2 above)
Comment (Mikehrn) [reply packages/frontend-2/components/feedback/Dialog.vue]: Good point, updated 👍

File: packages/frontend-2/components/feedback/Dialog.vue (hunk 1 above)
Comment (Mikehrn) [reply packages/frontend-2/components/feedback/Dialog.vue:69]: In this case it's a feedback form regarding Speckle as a product, I think in this case maybe no harm as we do want to gather feedback no matter what deployment this is on?

**Review by fabis94 — APPROVED** [review]

**Pull request merged**